### PR TITLE
maven_artifact: Allow to specify a custom timeout

### PR DESCRIPTION
**Pull request by ilpianista**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

maven_artifact
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

Add a 'timeout' parameter which allow to set a timeout in seconds for the requests. Defaults to 10 (as previously used by fetch_url).

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ilpianista for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ilpianista** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @ilpianista. To the current maintainers, @chrisisbeef please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: packaging/language/maven_artifact.py [hunk 1]
```diff
@@ -240,12 +246,14 @@ def _request(self, url, failmsg, f):
                 client = boto3.client('s3',aws_access_key_id=self.module.params.get('username', ''), aws_secret_access_key=self.module.params.get('password', ''))
                 url_to_use = client.generate_presigned_url('get_object',Params={'Bucket':bucket_name,'Key':key_name},ExpiresIn=10)
 
+        req_timeout = self.module.params.get('timeout', '10')
```
Comment (resmo) [inline packaging/language/maven_artifact.py]: the alternative value `'10'` should either be an int or even be removed as there is already a default in the arg spec.

**Comment by mattclay** [comment]
Tests on shippable failed because this PR branch was created before shippable was enabled. If you rebase the branch the tests will pass, although in this case the tests will just no-op since there are no tests for this module.

**Comment by ilpianista** [comment]
@chrisisbeef ping

**Comment by chrisisbeef** [comment]
shipit

Thanks @ilpianista

**Comment by gregdek** [comment]
Thanks again to @ilpianista for this PR, and thanks @chrisisbeef for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ilpianista** [comment]
Up!

**Pull request merged**